lessons/5.mdx

```diff
@@ -0,0 +1,1362 @@
+a direct import from `RainbowKit` that is generated using the configurations
+setup previously in our Main Application.
+
+With this setup we can run the `yarn dev` command in our console and click 
+the returned link (`http://localhost:portnumber`) to preview our application. 
+This is a good way to debug and see if our code is behaving as expected.
+
+The preview should look like this: 
+
+![App Header Preview](/assets/lessons/5/img_1.png)
+
+### NFT Cards
+
+Next we need to create some cards that display the 3 tiers of NFTs we created
+in our smart contract and provide a way to mint them.
+
+```jsx
+export default function Home() {
+  // variable definitions...
+
+  const {
+      data: mintData,
+      writeAsync: mint,
+      isLoading: isMintLoading,
+    } = useContractWrite({
+      addressOrName: CONTRACT_ADDRESS,
+      contractInterface: TierABI.abi,
+      functionName: "mint",
+    });
+
+ // return statement for rendering... 
+
+}
+```
+
+The [`useContractWrite`](https://wagmi.sh/docs/hooks/useContractWrite) hook 
+from WAGMI let's us pass in the contract address, contract ABI and the 
+function name to return a JavaScript function (that we have called `mint`) 
+that can interact with our smart contract and trigger a `write` function in it. 
+
+How cool is that?!
+
+It also returns some additional features for error handling and loading 
+management.
+We are assigning curated names to the returned values for easily calling them 
+later. For eaxmple, the `data` returned is called `mintData`.
+
+Even though we have this function, we need to pass in custom `message values` 
+i.e. an amount in **ETH**, as defined in our contract. For this we create our 
+own `mintToken` function that calls the `mint` function from WAGMI within it.
+
+```jsx
+// variable definitions and WAGMI write function...
+
+const mintToken = async (e) => {
+    try {
+      let mintTxn = await mint({
+        recklesslySetUnpreparedOverrides: {
+          value: ethers.utils.parseEther(e.target.value),
+        }
+      });
+      await mintTxn.wait();
+      console.log("This is the mint data", mintData);
+    } catch (error) {
+      console.log("Error minting NFT", error.message);
+    }
+  };
+
+// function Home code...
+```
+
+- We pass in `e` as an argument so that we can pick the amount of ETH 
+associated with a particular and use that for calling the mint function.
+- We want to wait for our minting transaction to complete before moving 
+on to other pieces of code.
+- After minting is processed we log the returned data to ensure we the 
+success of our transaction.
+- Using `trycatch` statements makes our error handling more robust. The 
+compiler tries each line of code and exits in between in an error occurs.
+We can console log this error to easily identify the cause and location 
+of the error with good commenting.
+
+Then we create a new folder named `nfts` in the `/public` directory and
```

Comment:
Then we create a new folder named `nfts` in the `/public` directory for saving our Tier SVGs. You can find links for the SVGs here.